src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/BalanceCalculatorUpdateActionTest.java

```diff
@@ -0,0 +1,128 @@
+                                                        String shiftStartTime,
+                                                        String shiftEndTime,
+                                                        BigDecimal expectedCumulativeTotal1,
+                                                        BigDecimal expectedCumulativeTotal2,
+                                                        BigDecimal expectedCumulativeTotal3,
+                                                        BigDecimal expectedCumulativeTotal4,
+                                                        BigDecimal expectedContributionsTotal1,
+                                                        BigDecimal expectedContributionsTotal2,
+                                                        BigDecimal expectedContributionsTotal3,
+                                                        BigDecimal expectedContributionsTotal4)
+      throws IOException {
+
+    List<AccrualModule> accrualModules = List.of(new AnnualTargetHoursAccrualModule());
+    ContributionsHandler contributionsHandler = new ContributionsHandler(accrualModules);
+    BalanceCalculator balanceCalculator = new BalanceCalculator(accrualsService,
+        contributionsHandler);
+
+    TimeEntry timeEntry = CommonUtils.createTimeEntry(timeEntryId, PERSON_ID, shiftStartTime,
+        shiftEndTime);
+
+    String tenantId = timeEntry.getTenantId();
+
+    when(accrualsService.getApplicableAgreement(tenantId, PERSON_ID,
+        timeEntry.getActualEndTime().toLocalDate()))
+        .thenReturn(loadObjectFromFile("data/agreement.json", Agreement.class));
+
+    when(accrualsService.getImpactedAccruals(tenantId, PERSON_ID, timeEntryId,
+        timeEntry.getActualStartTime().toLocalDate(), timeEntry.getActualEndTime().toLocalDate()))
+        .thenReturn(loadAccrualsFromFile("data/accruals_annualTargetHours.json"));
+
+    List<Accrual> accruals = balanceCalculator.calculate(timeEntry, KafkaAction.UPDATE);
+
+    assertThat(accruals).hasSize(4);
+
+    assertCumulativeTotal(accruals.get(0), expectedCumulativeTotal1);
+    assertCumulativeTotal(accruals.get(1), expectedCumulativeTotal2);
+    assertCumulativeTotal(accruals.get(2), expectedCumulativeTotal3);
+    assertCumulativeTotal(accruals.get(3), expectedCumulativeTotal4);
+
+    assertContributionsTotal(accruals.get(0), expectedContributionsTotal1);
+    assertContributionsTotal(accruals.get(1), expectedContributionsTotal2);
+    assertContributionsTotal(accruals.get(2), expectedContributionsTotal3);
+    assertContributionsTotal(accruals.get(3), expectedContributionsTotal4);
+
+  }
+
+  private void assertContributionsTotal(Accrual accrual, BigDecimal expectedContributionsTotal) {
+    assertThat(accrual.getContributions().getTotal()).usingComparator(
+        BigDecimal::compareTo).isEqualTo(expectedContributionsTotal);
+  }
+
+
+  private void assertCumulativeTotal(Accrual accrual, BigDecimal expectedCumulativeTotal) {
```

Comment:
merge into a single method and align this with similar private method in the Create Action integration test. Looks like a candidate for a test utils method

Reply:
Method moved to commonUtils. With Date assertion added.

Reply:
I can't see this 🤔

Reply:
Updated Create, Delete, Update action test classes to use new utils method